File: sdk/python/pe/resources/pa.py
import httpx
from typing import Optional


class PAClient:
    def __init__(self, http: httpx.Client):
        self._http = http

    def payers(self) -> dict:
        return self._http.get("/pa/payers").json()

    def submit(self, payer_id: str, *, patient_ref: Optional[str] = None,
               icd10: Optional[str] = None, cpt_code: Optional[str] = None,
               scenario: Optional[str] = None) -> dict:
        body = {"payerId": payer_id}
        if patient_ref:
            body["patientRef"] = patient_ref
        if icd10:
            body["icd10"] = icd10
        if cpt_code:
            body["cptCode"] = cpt_code
        if scenario:
            body["scenario"] = scenario
        return self._http.post("/pa/submit", json=body).json()

    def get(self, id: str) -> dict:
        return self._http.get(f"/pa/{id}").json()

    def submit_info(self, id: str, additional_info: str) -> dict:
        return self._http.post(f"/pa/{id}/info", json={"additionalInfo": additional_info}).json()

    def appeal(self, id: str, reason: str, scenario: Optional[str] = None) -> dict:
        body: dict = {"reason": reason}
        if scenario:
            body["scenario"] = scenario
        return self._http.post(f"/pa/{id}/appeal", json=body).json()

    def timeline(self, id: str) -> dict:
        return self._http.get(f"/pa/{id}/timeline").json()

File: packages/sdk/src/resources/pa.ts
import type { HttpClient } from '../http';
import type { PASimulation, PayerProfile } from '../types';

export interface SubmitPAOptions {
  payerId: string;
  patientRef?: string;
  icd10?: string;
  cptCode?: string;
  scenario?: string;
}

export class PAClient {
  constructor(private readonly http: HttpClient) {}

  payers(): Promise<{ payers: PayerProfile[] }> {
    return this.http.get('/pa/payers');
  }

  submit(opts: SubmitPAOptions): Promise<PASimulation> {
    return this.http.post<PASimulation>('/pa/submit', opts);
  }

  get(id: string): Promise<PASimulation> {
    return this.http.get<PASimulation>(`/pa/${id}`);
  }

  submitInfo(id: string, additionalInfo: string): Promise<PASimulation> {
    return this.http.post<PASimulation>(`/pa/${id}/info`, { additionalInfo });
  }

  appeal(id: string, reason: string, scenario?: string): Promise<PASimulation> {
    return this.http.post<PASimulation>(`/pa/${id}/appeal`, { reason, ...(scenario ? { scenario } : {}) });
  }

  timeline(id: string): Promise<{ id: string; currentStatus: string; timeline: PASimulation['timeline']; total: number }> {
    return this.http.get(`/pa/${id}/timeline`);
  }
}
